src/Core/Ast/AstMap/AstInherit.php

```diff
@@ -54,10 +58,6 @@ public function __toString(): string
      */
     public function withPath(array $path): self
     {
-        $self = clone $this;
-
-        $self->path = $path;
-
-        return $self;
+        return new self($this->classLikeName, $this->fileOccurrence, $this->type, $path);
```

Comment:
This, in combination with the new constructor argument, feels a bit off to me. I would think that `withPath()` adds the paths to the existing ones instead of replacing them, so maybe rename to `replacePath(array $path)`.

Maybe this is something for a separate PR where we make the constructor private and then have named factory methods. That way we can also encapsulate the AstInheritType instead of passing it, e.g. something like:

```
public static function fromExtends(ClassLikeToken $token, FileOccurence $occurence, array $path = []): self
{
     return new self($token, $occurence, AstInheritType::EXTENDS, $path);
}
```

Reply:
Replaced with `replacePath(array $path)`